packages/web/src/pages/sign-up-page/pages/PickHandlePage.tsx

```diff
@@ -1,33 +1,94 @@
-import { useCallback } from 'react'
+import { useCallback, useEffect, useRef } from 'react'
 
-import { Button } from '@audius/harmony'
-import { Formik, Form } from 'formik'
-import { useDispatch } from 'react-redux'
+import { Box, Button, Flex, Text } from '@audius/harmony'
+import { Form, Formik, FormikProps, useFormikContext } from 'formik'
+import { debounce, isEmpty } from 'lodash'
+import { useDispatch, useSelector } from 'react-redux'
+import { toFormikValidate, toFormikValidationSchema } from 'zod-formik-adapter'
 
 import { setValueField } from 'common/store/pages/signon/actions'
-import { TextField } from 'components/form-fields'
+import { getHandleField } from 'common/store/pages/signon/selectors'
+import { HarmonyTextField } from 'components/form-fields/HarmonyTextField'
 import { useNavigateToPage } from 'hooks/useNavigateToPage'
+import TwitterLogin from 'react-twitter-auth'
+import { audiusBackendInstance } from 'services/audius-backend/audius-backend-instance'
 import { SIGN_UP_FINISH_PROFILE_PAGE } from 'utils/route'
+import {
+  errorMessages as handleErrorMessages,
+  handleSchema
+} from '../utils/handleSchema'
+import styles from './PickHandlePage.module.css'
 
 const messages = {
-  header: 'Pick Your Handle',
-  description:
+  pickYourHandle: 'Pick Your Handle',
+  outOf: 'of',
+  handleDescription:
     'This is how others find and tag you. It is totally unique to you & cannot be changed later.',
   handle: 'Handle',
-  continue: 'Continue'
-}
-
-const initialValues = {
-  handle: ''
+  continue: 'Continue',
+  linkToClaim: 'Link to claim.',
+  ...handleErrorMessages
 }
 
 type PickHandleValues = {
   handle: string
 }
 
+const HandleField = () => {
+  const {
+    values,
+    validateForm,
+    errors: { handle: error },
+    touched
+  } = useFormikContext<PickHandleValues>()
+  const debouncedValidate = useCallback(debounce(validateForm, 1000), [
+    validateForm
+  ])
+  useEffect(() => {
+    debouncedValidate(values)
+  }, [values.handle])
+
+  let helperText: React.ReactNode = error
+
+  {
+    /* TODO: Finish this + other social linking: */
+  }
+  if (error === messages.twitterReservedError) {
+    helperText = (
+      <>
+        {messages.twitterReservedError}
+        <TwitterLogin
+          onFailure={() => {}}
+          onSuccess={() => {}}
+          requestTokenUrl={`${audiusBackendInstance.identityServiceUrl}/twitter`}
+          loginUrl={`${audiusBackendInstance.identityServiceUrl}/twitter/callback`}
+          // @ts-expect-error
```

Comment:
Should we just allow className prop?

Reply:
this is a 3rd party component! but the classname works